setup.py

```diff
@@ -19,6 +20,9 @@
 
 python_version = platform.python_version().split('.')[0]
 
+is_venv = (hasattr(sys, 'real_prefix') or\
+            (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix))
+
```

Comment:
I believe this works but it's a strange read. Actually entering the vend sets an environment variable:

```
$VIRTUAL_ENV
```

That is only present if you have activated the virtual environment. I saw reports that deactivating the venv still leaves that variable in the environment but that is not true from my tests. Thus I expect this to have been fixed in python. As we do care less on py2 I think checking for the environment variable might be an easier read, unless you have objections why that should be avoided ?

Reply:
I have not coded it like that because I also read similar comments about not doing it as you did. I was also considering not including py2 into the equation. In fact the check on `real_prefix` is doen for `virtualenv` module from python2 and the other check `base_prefix` is done for `venv` module from python3. 

So if we do not care about python2 we could remove the `real_prefix` verification and then we would be using the appropriate check then. I verified this `base_prefix` check on the documentation [here](https://docs.python.org/3/library/sys.html#sys.base_prefix). Probably the  `hasattr(sys, 'base_prefix')` part can also be dropped as this is for python3 < 3.3 (the oldest python3 maintained upstream is 3.5).

So if we align with python3 upstream having

```python
is_venv = sys.base_prefix != sys.prefix
```
should be enough. I am fine reducing it to the above. I could also add a comment stating that `sys.prefix` points to the venv path, if present, and that `sys.base_prefix` points always to the prefix used during python compilation.

Reply:
I think this would be good :+1: